src/main/java/uk/gov/hmcts/divorce/common/event/SwitchedToSoleFinalOrder.java

```diff
@@ -117,7 +122,13 @@ public SubmittedCallbackResponse submitted(CaseDetails<CaseData, State> details,
 
         log.info("SWITCH_TO_SOLE_FO submitted callback invoked for case id: {}", details.getId());
 
-        notificationDispatcher.send(switchedToSoleFoNotification, details.getData(), details.getId());
+        CaseData caseData = details.getData();
+
+        notificationDispatcher.send(switchedToSoleFoNotification, caseData, details.getId());
+
+        if (YesOrNo.YES.equals(caseData.getFinalOrder().getIsFinalOrderOverdue())) {
```

Comment:
You don't need this check here, it is already inside GeneralReferralService

Reply:
I forgot, ill revert.